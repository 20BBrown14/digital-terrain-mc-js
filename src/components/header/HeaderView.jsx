import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';
import { Menu } from 'antd';
import {
  NK_HOME,
  NK_GALLERY,
  NK_MAP,
  NK_RULES,
  NK_SERVER_INFORMATION,
  NK_APPLY,
  NK_ABOUT_US,
  NK_LOGIN,
} from '../../constants/navKeys';
import selectedNavKeyPropTypes from '../../proptypes/selectedNavKeyPropTypes';
import animatedBee from '../../assets/animated_bee.gif';

const propTypes = {
  /* The page to load base on the requested url from server */
  selectedNavKey: selectedNavKeyPropTypes.isRequired,
  /* Function for handling menu item click */
  handleNavMenuClick: PropTypes.func.isRequired,
};

const beeIcon = <img src={animatedBee} alt="loading..." className="header-icon" />;

/**
 * Header view. Renders menu items.
 */
const HeaderView = (props) => {
  const { selectedNavKey, handleNavMenuClick } = props;
  return (
    <div>
      <Menu
        selectedKeys={[selectedNavKey]}
        mode="horizontal"
        className="header-menu"
        onClick={(event) => { handleNavMenuClick(event.key); }}
      >
        <Menu.Item
          disabled
          key="icon"
          icon={beeIcon}
          className="header-icon-menu-item"
        />
        <Menu.Item
          key="title"
          className="header-title header-non-highlighted-menu-item"
        >
          Digital Terrain
        </Menu.Item>
        <Menu.Item
          key={NK_HOME}
        >
          Home
        </Menu.Item>
        <Menu.Item
          key={NK_GALLERY}
        >
          Gallery
        </Menu.Item>
        <Menu.Item
          key={NK_MAP}
        >
          Map
        </Menu.Item>
        <Menu.Item
          key={NK_RULES}
        >
          Rules
        </Menu.Item>
        <Menu.Item
          key={NK_SERVER_INFORMATION}
        >
          Server Information
        </Menu.Item>
        <Menu.Item
          key={NK_APPLY}
        >
          Apply
        </Menu.Item>
        <Menu.Item
          key={NK_ABOUT_US}
        >
          About us
        </Menu.Item>
        <Menu.Item
          key={NK_LOGIN}
        >
          <a href="https://discord.com/api/oauth2/authorize?client_id=693123471210709072&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Floggedin&response_type=token&scope=identify%20guilds">Login</a>
        </Menu.Item>
      </Menu>
    </div>
  );
};

HeaderView.propTypes = propTypes;

export default HeaderView;
