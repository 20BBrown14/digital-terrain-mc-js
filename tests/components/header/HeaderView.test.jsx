import React from 'react';
import HeaderView from '../../../src/components/header/HeaderView';

describe('HeaderView', () => {
  let view;
  beforeEach(() => {
    view = shallow(
      <HeaderView
        selectedNavKey="home"
        handleNavMenuClick={() => {}}
        showTitle={true}
      />
    );
  });

  it('renders a default view', () => {
    view = render(
      <HeaderView
        selectedNavKey="home"
        handleNavMenuClick={() => {}}
        showTitle={true}
      />
    );
    expect(view).toMatchSnapshot();
  });

  it('has a menu', () => {
    const menu = view.find('Menu');
    const menuItems = view.find('MenuItem');
    expect(menu).toHaveLength(1);
    expect(menuItems).toHaveLength(10);
  });

  it('calls nav menu click handler when menu is clicked', () => {
    const mockHandleNavMenuClick = jest.fn();
    view = shallow(
      <HeaderView
        selectedNavKey="home"
        handleNavMenuClick={mockHandleNavMenuClick}
        showTitle={true}
      />
    );
    view.find('Menu').at(0).simulate('click', { key: 'map' });
    expect(mockHandleNavMenuClick).toHaveBeenCalledTimes(1);
    expect(mockHandleNavMenuClick).toHaveBeenCalledWith('map');
  });
});
