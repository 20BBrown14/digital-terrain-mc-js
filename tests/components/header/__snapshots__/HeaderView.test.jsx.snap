// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderView renders a default view 1`] = `
<div>
  <ul
    class="ant-menu header-menu ant-menu-light ant-menu-root ant-menu-horizontal"
    role="menu"
  >
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      aria-disabled="true"
      class="ant-menu-item header-icon-menu-item ant-menu-item-only-child ant-menu-item-disabled"
      role="menuitem"
    >
      <img
        alt="loading..."
        class="header-icon"
        src="test-file-stub"
      />
      <span />
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item header-title header-non-highlighted-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Digital Terrain
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
      role="menuitem"
    >
      Home
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Gallery
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Map
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Rules
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Server Information
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Apply
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      About us
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="visibility:hidden;position:absolute"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
</div>
`;
